platform/darwin/src/MGLOpenGLStyleLayer.mm

```diff
@@ -107,13 +68,20 @@ - (instancetype)initWithIdentifier:(NSString *)identifier {
     return (mbgl::style::CustomLayer *)super.rawLayer;
 }
 
+- (MGLMapView *)mapView {
+    if ([self.style.stylable isKindOfClass:[MGLMapView class]]) {
```

Comment:
As implemented in this PR, MGLOpenGLStyleLayer doesn’t support snapshotting, but it wouldn’t be too difficult to make it work if necessary.